src/common/config_opts.h

```diff
@@ -318,6 +318,7 @@ OPTION(mon_crush_min_required_version, OPT_STR, "firefly")
 OPTION(mon_warn_on_crush_straw_calc_version_zero, OPT_BOOL, true) // warn if crush straw_calc_version==0
 OPTION(mon_warn_on_osd_down_out_interval_zero, OPT_BOOL, true) // warn if 'mon_osd_down_out_interval == 0'
 OPTION(mon_warn_on_cache_pools_without_hit_sets, OPT_BOOL, true)
+OPTION(mon_warn_osd_usage_percent, OPT_FLOAT, .40) // warn if difference in usage percent between OSDs exceeds specified percent
```

Comment:
mon_warn_osd_usage_percent is a new config parameter that warns when OSDs aren't being evenly used.  This can be caused by many different things like not enough PGs, bad weighting or even storing non-OSD data on an OSD disk.  Being aware of this situation could allow someone to take action before the most full OSD runs out of space and crashes while the cluster seems to have lots of space available over all.

mon_osd_min_in_ratio was the config parameter that was increased.  Say that if a network failure causes the cluster to partition and a significant part of the cluster goes down.  If we allow the automatic mechanism to mark too many of the OSDs _out_ an attempt to recover with the remaining nodes could result in running out of space.